File: Tests/RepoPromptCodeMapCoreTests/Fixtures/java/smoke.java
package example;

import java.util.List;

interface TaskRepository {
    List<Task> all();
    void save(Task task);
}

class Task {
    private final String id;
    private String title;

    Task(String id, String title) {
        this.id = id;
        this.title = title;
    }

    String label() {
        return id + ":" + title;
    }

    void rename(String title) {
        this.title = title;
    }
}

public class TaskService {
    private final TaskRepository repository;

    public TaskService(TaskRepository repository) {
        this.repository = repository;
    }

    public List<Task> load() {
        return repository.all();
    }
}

File: Tests/RepoPromptCodeMapCoreTests/Fixtures/rb/smoke.rb
require "json"

class Task
  DEFAULT_STATUS = "open"

  attr_reader :title

  def initialize(title)
    @title = title
  end

  def rename(next_title)
    @title = next_title
  end

  def self.from_json(json)
    new(JSON.parse(json)["title"])
  end
end

module TaskFormatting
  def self.label(task)
    task.title.upcase
  end
end

def build_task(title)
  Task.new(title)
end
